cmd/manager/main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/infrakit/cli"
+	"github.com/docker/infrakit/discovery"
+	"github.com/docker/infrakit/leader"
+	"github.com/docker/infrakit/manager"
+	"github.com/docker/infrakit/rpc"
+	group_rpc "github.com/docker/infrakit/rpc/group"
+	"github.com/docker/infrakit/store"
+	"github.com/spf13/cobra"
+)
+
+type backend struct {
+	id       string
+	plugins  discovery.Plugins
+	leader   leader.Detector
+	snapshot store.Snapshot
+}
+
+func main() {
+
+	logLevel := cli.DefaultLogLevel
+
+	backend := &backend{
+		id: "group",
+	}
```

Comment:
I think the shared mutable state here makes the code harder to read than it should be.  Can you instead convert the `PersistentPostRunE` to a function you call and pass a `backend`?